packages/plugin-ext/src/main/browser/languages-main.ts

```diff
@@ -100,23 +106,32 @@ export class LanguagesMainImpl implements LanguagesMain {
     $registerCompletionSupport(handle: number, selector: SerializedDocumentFilter[], triggerCharacters: string[], supportsResolveDetails: boolean): void {
         this.disposables.set(handle, monaco.modes.CompletionProviderRegistry.register(fromLanguageSelector(selector), {
             triggerCharacters,
-            provideCompletionItems: (model, position, context, token) =>
-                this.proxy.$provideCompletionItems(handle, model.uri, position, context, token).then(result => {
-                    if (!result) {
-                        return undefined;
-                    }
-                    return {
-                        suggestions: result.completions,
-                        incomplete: result.incomplete,
-                        dispose: () => this.proxy.$releaseCompletionItems(handle, result.id)
-                    };
-                }),
+            provideCompletionItems: (model, position, context, token) => this.provideCompletionItemsImpl(handle, model, position, context, token),
             resolveCompletionItem: supportsResolveDetails
-                ? (model, position, suggestion, token) => Promise.resolve(this.proxy.$resolveCompletionItem(handle, model.uri, position, suggestion, token))
+                ? (model, position, suggestion, token) => Promise.resolve(this.resolveCompletionItemImpl(handle, model, position, suggestion, token))
                 : undefined
         }));
     }
 
+    protected provideCompletionItemsImpl(handle: number, model: monaco.editor.ITextModel, position: monaco.Position,
```

Comment:
I don't think we need `Impl` suffix, it reads strange like it provides an implementation for completion items. What was the reasoning behind such naming? Could we drop the suffix?